bindings/go/src/fdb/database.go

```diff
@@ -157,19 +159,47 @@ func retryable(wrapped func() (interface{}, error), onError func(Error) FutureNi
 // See the Transactor interface for an example of using Transact with
 // Transaction and Database objects.
 func (d Database) Transact(f func(Transaction) (interface{}, error)) (interface{}, error) {
+	return d.transact(context.Background(), f)
+}
+
+// TransactWithContext - same as Transact but uses context for setting timeout
+func (d Database) TransactWithContext(ctx context.Context, f func(Transaction) (interface{}, error)) (interface{}, error) {
+	return d.transact(ctx, f)
+}
+
+func (d Database) transact(ctx context.Context, f func(Transaction) (interface{}, error)) (interface{}, error) {
 	tr, e := d.CreateTransaction()
 	// Any error here is non-retryable
 	if e != nil {
 		return nil, e
 	}
 
+	if ctx != nil {
+		if deadline, ok := ctx.Deadline(); ok {
+			t := time.Until(deadline).Milliseconds()
+			_ = tr.Options().SetTimeout(t)
+		}
+
+		go func() {
+			<-ctx.Done()
```

Comment:
It seems when we call old `Transact` which takes `context.Background()`,  it will never never end, as `context.Background()` isn't cancellable by default IIUC. https://go.dev/play/p/k2gS841pcuH